src/main/java/com/thealgorithms/datastructures/trees/SplayTree.java

```diff
@@ -0,0 +1,296 @@
+                root.left.left = splay(root.left.left, key);
+                // Perform zig operation on parent
+                root = rotateRight(root);
+            } // Zig-Zag case
+            else if (root.left.key < key) {
+                // Recursive call to splay on grandchild
+                root.left.right = splay(root.left.right, key);
+                // Perform zag operation on parent
+                if (root.left.right != null) root.left = rotateLeft(root.left);
+            }
+
+            return (root.left == null) ? root : rotateRight(root);
+        } else {
+            if (root.right == null) return root;
+            // Zag-Zag case
+            if (root.right.key > key) {
+                // Recursive call to splay on grandchild
+                root.right.left = splay(root.right.left, key);
+                // Perform zig operation on parent
+                if (root.right.left != null) root.right = rotateRight(root.right);
+            } // Zag-Zig case
+            else if (root.right.key < key) {
+                // Recursive call to splay on grandchild
+                root.right.right = splay(root.right.right, key);
+                // Perform zag operation on parent
+                root = rotateLeft(root);
+            }
+
+            return (root.right == null) ? root : rotateLeft(root);
+        }
+    }
+
+    /**
+     * Insert a key into the SplayTree.
+     *
+     * @param key The key to insert.
+     */
+    public void insert(int key) {
+        root = insertRec(root, key);
+        root = splay(root, key);
+    }
+
+    /**
+     * Recursive function to insert a key into a subtree.
+     *
+     * @param root The root of the subtree to insert the key into.
+     * @param key  The key to insert.
+     * @return The root of the modified subtree after insertion.
+     * @throws IllegalArgumentException If the key to be inserted already exists in the subtree.
+     */
+    private Node insertRec(Node root, int key) {
+        if (root == null) return new Node(key);
+
+        if (key < root.key) {
+            root.left = insertRec(root.left, key);
+        } else if (key > root.key) {
+            root.right = insertRec(root.right, key);
+        } else {
+            throw new IllegalArgumentException("Duplicate key: " + key);
+        }
+
+        return root;
+    }
+
+    /**
+     * Search for a key in the SplayTree.
+     *
+     * @param key The key to search for.
+     * @return True if the key is found, otherwise false.
+     */
+    public boolean search(int key) {
+        root = splay(root, key);
+        return root != null && root.key == key;
+    }
+
+    /**
+     * Deletes a key from the SplayTree.
+     *
+     * @param key The key to delete.
+     * @throws IllegalArgumentException If the tree is empty.
+     */
+    public void delete(int key) {
+        if (root == null) {
+            throw new IllegalArgumentException("Cannot delete from an empty tree");
+        }
```

Comment:
Please add an `isEmpty` method. Otherwise using this class might be very annoying.

Reply:
I think we can define the `isEmpty` method and use it to check the entire tree's `root` and continue using `root == null` checks within recursive calls because `root` is a local variable that represents the current node being processed in the recursion.